### PR TITLE
Expose the office name in the dashboard

### PR DESCRIPTION
**Pull request by zspencer**
Reason for Change
===================
* When working on another ticket, I had noticed that we said "Michigan Bridges" as the recipient which is... annoying at best and misleading at worst. I wanted to make sure we could more easily answer the question of "Which office did the fax go to?" and "Which office did should the fax go to?"

##### Before
screen shot 2017-09-14 at 1 18 57 pm

##### After
screen shot 2017-09-14 at 1 32 50 pm

### CODE REVIEW

File: app/models/fax_recipient.rb [hunk 1]
```diff
@@ -12,12 +12,23 @@ def number
   end
 
   def name
-    ENV.fetch("FAX_RECIPIENT", "Michigan Bridges")
+    office["name"].to_s.titleize
+  end
+
+  def office
```
Comment (jessieay) [inline app/models/fax_recipient.rb]: seems like this could use a unit test

File: app/models/fax_recipient.rb [hunk 2]
```diff
@@ -12,12 +12,23 @@ def number
...
+      office
+    else
+      self.class.find_by(zip: residential_address.zip)
+    end
   end
 
   def self.find_by(zip:)
     name = covered_zip_codes[zip] || configuration["fallback_office"]
     Rails.logger.debug("Mapped zip #{zip} to office #{name}")
+    offices[name]["name"] = name
```
Comment (jessieay) [inline app/models/fax_recipient.rb]: this feels a bit confusing -- why this change?

**Comment by alanjosephwilliams** [comment]
great catch and significant mprovement

File: app/models/snap_application.rb [hunk 3]
```diff
@@ -65,6 +65,18 @@ def residential_address
     addresses.where.not(mailing: true).first || NullAddress.new
   end
 
+  def zip
+    residential_address.zip
+  end
+
+  def receiving_office
```
Comment (jessieay) [inline app/models/snap_application.rb]: thoughts on putting this below `receiving_office_name` so that the methods in this file are in dependence order? (methods are above other methods that they depend on). This is something I picked up from Avdi Grimm's ruby tapas!

File: app/models/fax_recipient.rb (hunk 2 above)
Comment (zspencer) [reply app/models/fax_recipient.rb]: This was hella annoying. It's because our yml file doesn't include the name of the office in the data, just as the key. It would be better to do as part of processing the YML file initially, not this far downstream.

File: app/models/snap_application.rb (hunk 3 above)
Comment (zspencer) [reply app/models/snap_application.rb]: Nice call! I wish there was a rubocop for that!

File: app/models/fax_recipient.rb (hunk 2 above)
Comment (jessieay) [reply app/models/fax_recipient.rb]: interesting -- so is there a test we can write that confirms that change does what we want?

**Comment by zspencer** [comment]
@jessieay  - Resolved your feedback, I think?

File: config/offices.yml [hunk 4]
```diff
@@ -2,7 +2,7 @@
 defaults: &defaults
   # Which office to send the fax to in the event the submitted
   # zip code is not in our mapping
-  fallback_office: "union"
+  fallback_office: "clio"
```
Comment (jessieay) [inline config/offices.yml]: good change! the different receiving office fallback in different envs was confusing to me when I was working on the `office_location` stuff

File: spec/models/fax_recipient_spec.rb [hunk 5]
```diff
@@ -1,135 +1,77 @@
...
+        snap_application: app(from_zip_covered_by: :testing),
+      )
 
-        expect(fax_recipient.number).to eq clio_fax_number
-      end
+      expect(fax_recipient.name).to eq fax_recipient.office["name"]
     end
   end
-
   def sandbox_fax_number
```
Comment (jessieay) [inline spec/models/fax_recipient_spec.rb]: looks like we removed tests for sandbox fax number but did not re-add any?

Comment (zspencer) [reply spec/models/fax_recipient_spec.rb]: Yeah, I removed them because they weren't as important as the non-production ones, and forgot to clean out this method.

**Comment by zspencer** [comment]
[Improve admin overview by exposing which office we faxed the 1171 to](https://trello.com/c/5jKZ5tbM/313-improve-admin-overview-by-exposing-which-office-we-faxed-the-1171-to)
